controllers/space/MembershipController.php

```diff
@@ -68,4 +69,51 @@ public function actionIndex($spaceId){
         }
         return $this->returnPagination($query,$pagination,$results);
     }
+
+    /**
+     * @return void
+     * Get all the spaces list of the user
+     */
+    //Get the list of all spaces of the user (issue-8 new endpoint)
+    public function actionGetSpaces(){
+       $userId = Yii::$app->user->id;
+
+       $user = User::findOne($userId);
+
+       if($user == null){
+           return $this->returnError(404, 'User not found!');
+       }
+
+       $spaceMemberships = Membership::find()->where(['user_id' => $userId]);
+
+        if($spaceMemberships->all() == null){
+            return $this->returnError(404, 'Space not found for the user : '.$userId);
+        }
+
+        $results = [];
+        $pagination = $this->handlePagination($spaceMemberships);
+
+        foreach ($spaceMemberships->all() as $spaceMembership){
+            $results[] = $this->getSpacesListData($spaceMembership);
+        }
+
+        return $this->returnPagination($spaceMemberships,$pagination,$results);
+    }
+
+    public function getSpacesListData(Membership $membership){
+        return [
+            'user' => UserDefinitions::getUserShort($membership->user),
+            'space_id' => $membership->space_id,
```

Comment:
please make this `id` and also add the guid of the space as `guid`

Reply:
Okay @donni106 
We will add these keys